src/main/java/com/direwolf20/buildinggadgets/common/tools/gadget/placement/PlacementTarget.java

```diff
@@ -0,0 +1,29 @@
+package com.direwolf20.buildinggadgets.common.tools.gadget.placement;
+
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+
+public class PlacementTarget {
```

Comment:
This should be marked final, I'll use this in 1.13 to replace the old and strange BlockMap class.